pkg/operator/defaultscccontroller/defaultscc.go

```diff
@@ -0,0 +1,98 @@
+package defaultscccontroller
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+
+	"github.com/openshift/api"
+	securityv1 "github.com/openshift/api/security/v1"
+
+	assets "github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/defaultscc_assets"
+)
+
+// DefaultSCCCache maintains the list of default SecurityContextConstraints objects
+// shipped wth an OpenShift cluster.
+type DefaultSCCCache struct {
+	orderedNames []string
+	set          map[string]*securityv1.SecurityContextConstraints
+}
+
+// Get returns the SecurityContextConstraints object associated with the
+// specified name. The returned SecurityContextConstraints is a deep copy and so
+// it can be modified at will without impacting the integrity of the underlying cache.
+func (d *DefaultSCCCache) Get(name string) (scc *securityv1.SecurityContextConstraints, exists bool) {
+	obj, exists := d.set[name]
+	if !exists {
+		return
+	}
+
+	scc = obj.DeepCopy()
```

Comment:
would move this to the place where you mutate the SCC. Defensive DeepCopies are an anti-pattern.

Reply:
done.